suite/src/utils/ceramicHelpers.ts

```diff
@@ -31,32 +35,66 @@ function defaultMsgGenerator(stream: Stream) {
   )}`
 }
 
-async function withTimeout(prom: Promise<any>, timeoutSecs: number) {
+async function withTimeout(prom: Promise<any>, timeoutMs: number) {
   const startTime = new Date().toISOString()
   return new Promise((resolve, reject) => {
     setTimeout(() => {
       const curTime = new Date().toISOString()
       reject(
-        `Timed out after ${timeoutSecs} seconds. Current time: ${curTime}, start time: ${startTime}`,
+        `Timed out after ${timeoutMs} millis. Current time: ${curTime}, start time: ${startTime}`,
       )
-    }, timeoutSecs * 1000)
+    }, timeoutMs)
     prom.then(resolve)
   })
 }
 
+/**
+ * Loads a document from a ceramic node with a timeout.
+ * @param ceramicNode : CeramicClient to load the document from
+ * @param documentId : ID of the document to load
+ * @param timeoutMs : Timeout in milliseconds
+ * @returns The document if found, throws error on timeout
+ */
+export async function loadDocumentOrTimeout(
+  ceramicNode: CeramicClient,
+  documentId: StreamID,
+  timeoutMs: number,
+): Promise<Stream> {
+  let now = Date.now()
+  let count = 0
+  const expirationTime = now + timeoutMs
+  let lastError = null
+  while (now < expirationTime) {
+    try {
+      count += 1
+      return await ceramicNode.loadStream(documentId)
+    } catch (error) {
+      lastError = error
+      if (count % 10 === 0) {
+        console.log(`Error loading document : ${documentId} retrying`, error)
+      }
+      await delayMs(100)
```

Comment:
I think adjusting polling frequency dynamically might be a good change to land in this PR. Maybe it can go in the next one. We can start with a small delay and then adjust based on the number of retries. Might save us some cpu if we want to use this method with a larger value of timeoutMs. let me know your thoughts.
I was thinking something more in terms of : 

```
const initialDelay = 50;
const maxDelay = 500;
let currentDelay = initialDelay;

while (now < expirationTime) {
  try {
    count += 1;
    return await ceramicNode.loadStream(documentId);
  } catch (error) {
    lastError = error;
    if (count % 10 === 0) {
      console.log(`Error loading document : ${documentId} retrying`, error);
    }
    await delayMs(currentDelay);
   // Increase delay upto the max. value
    currentDelay = Math.min(currentDelay * 2, maxDelay); 
    now = Date.now();
  }
}


```

Reply:
tbh that feels like extra complexity for little gain.  Polling once every 100ms in a single thread is extremely cheap CPU-wise, even for production code that would be cheap, but this is test code which makes it even less important.  I'd rather keep things simple until/unless we notice a problem that leads us towards wanting to adjust this.